_posts/2020-12-02-spack-user-survey-2020.md

```diff
@@ -0,0 +1,918 @@
+top two U.S. systems, Summit and Sierra, are Power machines.
+
+We can't draw a fair comparison with EasyBuild on this question, as
+EasyBuild's survey asked users what CPUs they were *currently* using
+rather than what they expected to be using in the next year. Even if it's
+to be taken with a grain of salt, the difference is still worth
+mentioning. In the
+[EasyBuild survey](https://www.youtube.com/watch?v=ppKVsnwba7g#t=30m06s),
+the vast majority of users were similarly running on Intel machines. But,
+less than 20% of were using AMD chips, less than 5% were using Power, and
+only one user reported using ARM.
+
+## Which GPUs do you expect to use with Spack in the next year?
+
+<figure>
+  <a href="/assets/images/spack-user-survey-2020/two_multi_bars_gpus_next_year_any.svg">
+    <img src="/assets/images/spack-user-survey-2020/two_multi_bars_gpus_next_year_any.svg">
+  </a>
+</figure>
+
+All users in the Spack community expect to run on GPUs in the next year,
+and over 90% plan to build for NVIDIA GPUs. Around half of the overall
+community expects to build for AMD GPUs, and around 30% expect to build
+for Intel GPUs. Within ECP, the percentage of users planning to run on
+NVIDIA GPUs is very slightly lower, likely because NVIDIA will not be the
+GPU on any of the initial three exascale machines. Aurora will be an
+Intel GPU system and both Frontier and El Capitan will use AMD GPUs. As
+you might expect, 80% of ECP users expect to use AMD GPUs and over half
+expect to use Intel GPUs.
+
+[GPU usage in the EasyBuild community](https://www.youtube.com/watch?v=ppKVsnwba7g#t=31m17s)
+was similarly high -- 96.5% of EasyBuild users were compiling software
+for GPUs -- so it's pretty clear that GPUs have become pervasive in HPC.
+
+## Which compilers do you expect to use with Spack in the next year?
+
+<figure>
+  <a href="/assets/images/spack-user-survey-2020/two_multi_bars_compilers_next_year.svg">
+    <img src="/assets/images/spack-user-survey-2020/two_multi_bars_compilers_next_year.svg">
+  </a>
+</figure>
+
+As you might expect given the CPU and GPU results, Spack users
+anticipate using a very wide range of compilers. `gcc` is still king,
+with almost 100% of users expecting to use it, and LLVM and Intel
+compilers are next on the list.
+
+Interestingly, only around 60% planned to use `nvcc`, and a bit more than
+40% planned to use NVIDIA's HPC compilers. Given that over 90% of users
+said they expected to build on NVIDIA GPUs, we're tempted to explain the
+discrepancy by saying that a large percentage of users expect to use
+NVIDIA GPUs not through CUDA directly, but through GPU-optimized
+libraries or through compiler capabilities like OpenMP offload. The same
+can probably be said for AMD and Intel GPUs -- the percentage of users
+who plan to run with compilers specifically intended for these GPUs was
+consistently lower than the number of users who anticipated using them.
+
+## Rank upcoming Spack features by importance
+
+<figure>
```

Comment:
I'm on the fence about the colors in these graphs, which could be reversed. When I first scanned this graph, I thought Windows support was the most important feature. Obviously the opposite is true when reading more closely. Red can be interpreted as "critical" as in "high importance" (like an email with an exclamation mark) or "my progress is limited w/o this feature." But you're using green for "critical" to imply something like "the feature is in our work queue" or "the feature has the greenlight." The detailed explanation of the data is fine; I'm just saying the colors threw me off a bit for what they were showing. Color isn't as important in the other graphs, and the red-green spectrum in the "How would you rate the overall quality" section below totally makes sense.

Reply:
This is a good point.  Do you have an opinion on which color map we should use?  https://matplotlib.org/tutorials/colors/colormaps.html

Reply:
Both are currently using `RdYlGn`, but maybe a 2-color scale would be better here since it's more of a continuum and not positive vs. negative.

Reply:
Ok made this use blues.  I think we're done now.